### PR TITLE
add VP sample for scaling,csc,sharpness,denoise,chromasitting.

### PR DESCRIPTION
**Pull request by kelly-du**
Signed-off-by: kelly-du <kelly.du@intel.com>

### CODE REVIEW

File: videoprocess/Makefile.am [hunk 1]
```diff
@@ -37,6 +37,21 @@ TEST_LIBS = \
...
+vppdenoise_SOURCES = vppdenoise.cpp
+vppdenoise_LDADD   = $(TEST_LIBS)
+
+vppsharpness_SOURCES = vppsharpness.cpp
+vppsharpness_LDADD   = $(TEST_LIBS)
+
+vppchromasitting_SOURCES = vppchromasitting.cpp
+vppchromasitting_LDADD   = $(TEST_LIBS)
+
+vppscaling_n_out_usrptr_SOURCES = vppscaling_n_out_usrptr.cpp
```
Comment (jackie-guo) [inline videoprocess/Makefile.am]: Is this ready for check-in? Does this depend on libva related to change?

File: videoprocess/process.cfg.template [hunk 2]
```diff
@@ -39,8 +39,8 @@ FILTER_TYPE: VAProcFilterNone
 #default value will be applied then.
 
 #5.1 Denoise filter paramters
- #(0.0 ~ 1.0, default 0.5)
-DENOISE_INTENSITY: 0.75
+ #(0.0 ~ 64, default 0)
+DENOISE_INTENSITY: 20
```
Comment (jackie-guo) [inline videoprocess/process.cfg.template]: What is reason to use 20?  Sometimes DN effect is not so obvious, what about using a more stronger intensity to see obvious DN effect easily?

File: videoprocess/process.cfg.template [hunk 3]
```diff
@@ -54,8 +54,8 @@ DEINTERLACING_ALGORITHM:  VAProcDeinterlacingBob
 DEINTERLACING_FLAGS: 0
 
 #5.3 Sharpening parameters
- # (0.0 ~ 1.0, default 0.5)
-SHARPENING_INTENSITY: 0.75
+ # (0.0 ~ 64, default 44)
+SHARPENING_INTENSITY: 20
```
Comment (jackie-guo) [inline videoprocess/process.cfg.template]: Can we use 44 instead of 20?

File: videoprocess/process_denoise.cfg.template [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#2.Destination YUV(RGB) file information
+DST_FILE_NAME:    ./output_1920x1080.nv12
+DST_FRAME_WIDTH:  1920
+DST_FRAME_HEIGHT: 1080
+DST_FRAME_FORMAT: NV12
+
+DST_FILE_FORMAT: NV12
+
+#3.How many frames to be processed
+FRAME_SUM: 1
```
Comment (jackie-guo) [inline videoprocess/process_denoise.cfg.template]: Suggest to unify the frame no across these test applications, like 5 as previous chromasitting application.

File: videoprocess/process_denoise.cfg.template [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+#4.VPP filter specific parameters. If they are not specified here,
+#default value will be applied then.
+FILTER_TYPE: VAProcFilterNoiseReduction
+#4.1 Denoise filter paramters
+ #(0--64, default:0 )
+DENOISE_INTENSITY: 20
```
Comment (jackie-guo) [inline videoprocess/process_denoise.cfg.template]: Suggest use a bigger value to get more obvious DN effect.

File: videoprocess/process_scaling_csc.cfg.template [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+#2.Destination YUV(RGB) file information
+DST_FILE_NAME:    ./scaling_out_720x480.yv12
+DST_FRAME_WIDTH:  720
+DST_FRAME_HEIGHT: 480
+DST_FRAME_FORMAT: YV12
+
+DST_FILE_FORMAT: YV12
+
+#3.How many frames to be processed
+FRAME_SUM: 1
```
Comment (jackie-guo) [inline videoprocess/process_scaling_csc.cfg.template]: Suggest to unify the frame no, like 5 as previous chromasitting application.

File: videoprocess/process_chromasitting.cfg.template [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+# filter and the corresponding parameters.
+
+#1.Source YUV(RGB) file information
+#SRC_FILE_NAME:    /root/clips/YUV/bus_cif.yv12
+SRC_FILE_NAME: ./ChromaSittingTest_720x516.nv12
+SRC_FRAME_WIDTH: 720
+SRC_FRAME_HEIGHT: 516
+SRC_FRAME_FORMAT: NV12
+
+#Note .yv12 files are in YV12 format and .yuv used above for STDE is in I420 format
```
Comment (jackie-guo) [inline videoprocess/process_chromasitting.cfg.template]: Why STDE here?

File: videoprocess/process_denoise.cfg.template [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+# filter and the corresponding parameters.
+
+#1.Source YUV(RGB) file information
+#SRC_FILE_NAME:    /root/clips/YUV/bus_cif.yv12
+SRC_FILE_NAME: ./foreman_10f_640x480.nv12
+SRC_FRAME_WIDTH:  640
+SRC_FRAME_HEIGHT: 480
+SRC_FRAME_FORMAT: NV12
+
+#Note .yv12 files are in YV12 format and .yuv used above for STDE is in I420 format
```
Comment (jackie-guo) [inline videoprocess/process_denoise.cfg.template]: Why STDE here?

File: videoprocess/process_scaling_csc.cfg.template [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+#  you can modify this configuration file to set the corresponding parameters.
+
+#1.Source YUV(RGB) file information
+#SRC_FILE_NAME:    /root/clips/YUV/bus_cif.yv12
+SRC_FILE_NAME: ./foreman_10f_640x480.nv12
+SRC_FRAME_WIDTH:  640
+SRC_FRAME_HEIGHT: 480
+SRC_FRAME_FORMAT: NV12
+
+#Note .yv12 files are in YV12 format and .yuv used above for STDE is in I420 format
```
Comment (jackie-guo) [inline videoprocess/process_scaling_csc.cfg.template]: Why STDE here?

File: videoprocess/process_scaling_csc.cfg.template (hunk 6 above)
Comment (jackie-guo) [inline videoprocess/process_scaling_csc.cfg.template]: Suggest to unify the frame no, like 5 as previous chromasitting application.

File: videoprocess/process_sharpness.cfg.template [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+#2.Destination YUV(RGB) file information
+DST_FILE_NAME:    ./flowersky_352x288_writer352x288.nv12
+DST_FRAME_WIDTH:  352
+DST_FRAME_HEIGHT: 288
+DST_FRAME_FORMAT: NV12
+
+DST_FILE_FORMAT: NV12
+
+#3.How many frames to be processed
+FRAME_SUM: 3
```
Comment (jackie-guo) [inline videoprocess/process_sharpness.cfg.template]: Suggest to unify the frame no, like 5 as previous chromasitting application.

File: videoprocess/process_sharpness.cfg.template [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+# filter and the corresponding parameters.
+
+#1.Source YUV(RGB) file information
+#SRC_FILE_NAME:    /root/clips/YUV/bus_cif.yv12
+SRC_FILE_NAME: ./flowersky_352x288.nv12
+SRC_FRAME_WIDTH: 352 
+SRC_FRAME_HEIGHT: 288
+SRC_FRAME_FORMAT: NV12
+
+#Note .yv12 files are in YV12 format and .yuv used above for STDE is in I420 format
```
Comment (jackie-guo) [inline videoprocess/process_sharpness.cfg.template]: Why STDE here?

File: videoprocess/process_sharpness.cfg.template [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+#3.How many frames to be processed
+FRAME_SUM: 3
+
+#4.VPP filter specific parameters. If they are not specified here,
+#default value will be applied then.
+FILTER_TYPE: VAProcFilterSharpening
+
+#4.1 Sharpening parameters
+# (0 ~ 64, default 44)
+SHARPENING_INTENSITY: 16
```
Comment (jackie-guo) [inline videoprocess/process_sharpness.cfg.template]: Suggest to use 44.

File: videoprocess/Makefile.am (hunk 1 above)
Comment (kelly-du) [reply videoprocess/Makefile.am]: this PR not depend on libva changes

File: videoprocess/process.cfg.template (hunk 2 above)
Comment (kelly-du) [reply videoprocess/process.cfg.template]: > What is reason to use 20? Sometimes DN effect is not so obvious, what about using a more stronger intensity to see obvious DN effect easily?

I picked one lucas test to check the result, that case DN should be 20. I can change to a larger one

File: videoprocess/process.cfg.template (hunk 3 above)
Comment (kelly-du) [reply videoprocess/process.cfg.template]: > Can we use 44 instead of 20?

ok.

File: videoprocess/process_denoise.cfg.template (hunk 4 above)
Comment (kelly-du) [reply videoprocess/process_denoise.cfg.template]: > Suggest to unify the frame no across these test applications, like 5 as previous chromasitting application.
ok

File: videoprocess/process_denoise.cfg.template (hunk 5 above)
Comment (kelly-du) [reply videoprocess/process_denoise.cfg.template]: ok, change to 44.

File: videoprocess/process_scaling_csc.cfg.template (hunk 6 above)
Comment (kelly-du) [reply videoprocess/process_scaling_csc.cfg.template]: ok, change frame_sum to 5.

File: videoprocess/process_chromasitting.cfg.template (hunk 7 above)
Comment (kelly-du) [reply videoprocess/process_chromasitting.cfg.template]: removed the STDE comment

File: videoprocess/process_denoise.cfg.template (hunk 8 above)
Comment (kelly-du) [reply videoprocess/process_denoise.cfg.template]: removed the STDE comment

File: videoprocess/process_scaling_csc.cfg.template (hunk 9 above)
Comment (kelly-du) [reply videoprocess/process_scaling_csc.cfg.template]: removed the STDE comment

File: videoprocess/process_sharpness.cfg.template (hunk 10 above)
Comment (kelly-du) [reply videoprocess/process_sharpness.cfg.template]: fixed, change to 5

File: videoprocess/process_sharpness.cfg.template (hunk 11 above)
Comment (kelly-du) [reply videoprocess/process_sharpness.cfg.template]: fixed, remove the STDE comment

File: videoprocess/process_sharpness.cfg.template (hunk 12 above)
Comment (kelly-du) [reply videoprocess/process_sharpness.cfg.template]: fixed, change to 44

File: videoprocess/Makefile.am (hunk 1 above)
Comment (jackie-guo) [reply videoprocess/Makefile.am]: Here, you include vppscaling_n_out_usrptr.cpp, but I do not see vppscaling_n_out_usrptr.cpp is uploaded. Does vppscaling_n_out_usrptr.cpp depend on libva changes?

**Pull request merged**